pkg/cri/resource-manager/policy/builtin/balloons/balloons-policy.go

```diff
@@ -922,11 +927,31 @@ func (p *balloons) applyBalloonDef(balloons *[]*Balloon, blnDef *BalloonDef, fre
 	return nil
 }
 
+func (p *balloons) validateConfig(bpoptions *BalloonsOptions) error {
+	for _, blnDef := range bpoptions.BalloonDefs {
+		if blnDef.MaxCpus != NoLimit && blnDef.MinCpus > blnDef.MaxCpus {
+			return balloonsError("MinCpus (%d) > MaxCpus (%d) in balloon type %q",
+				blnDef.MinCpus, blnDef.MaxCpus, blnDef.Name)
+		}
+		if blnDef.Name == reservedBalloonDefName || blnDef.Name == defaultBalloonDefName {
+			continue
+		}
```

Comment:
just for my sake of knowledge, what it this for?

Reply:
It's a very valid question, @fmuyassarov .

There are two special balloon types which both have only one instance: the reserved balloon and the default balloon. The default balloon is sometimes almost an alias to the reserved balloon as it uses the same CPUs. But sometimes  it is not: it might have been configured explicitly, which then makes it a clearly separate balloon.

In this patch version, validations before this check are done to all balloon types (including the two built-in balloon types), while validations after this check only to user-configured balloon types.

However, your comment raises two questions:
1. If this kind of a separation is really wanted/needed, then it should be clearly commented in the code.
2. If this separation is not absolutely necessary (why built-in types shouldn't comply with all checks that are done to user-defined types), then why to have it.

I suspect that at some point I had some code that didn't really work with the built-in balloon types, and now that code does not exist anymore. Therefore I think I'll remove this check.

Thanks for pointing this out, @fmuyassarov !

Reply:
Fixed (removed).

Reply:
Thanks for detailed info @askervin. LGTM now.